obspy/io/mseed/util.py

```diff
@@ -653,25 +658,30 @@ def _parse_time(values):
                 "the maximum strictly allowed value is 9999. It will be "
                 "interpreted as one or more additional seconds." % values[5],
                 category=UserWarning)
-        return UTCDateTime(
-            year=values[0], julday=values[1],
-            hour=values[2], minute=values[3], second=values[4],
-            microsecond=msec % 1000000) + offset
+        try:
+            t = UTCDateTime(
+                year=values[0], julday=values[1],
+                hour=values[2], minute=values[3], second=values[4],
+                microsecond=msec % 1000000) + offset
+        except TypeError:
+            msg = 'Problem decoding time (wrong endian?)'
+            raise InternalMSEEDParseTimeError(msg)
```

Comment:
Maybe pass on the message of the exception which would ease debugging and improve understanding of whatever problem is encountered?

Reply:
I don't think the message would be very helpful, at least not in the case that seems to be what is normally encountered (when trying the wrong endian):

```
$ python -c 'from obspy import UTCDateTime; UTCDateTime(year=444444, julday=10)'
Version 7.000kness 20eader in AFM: UnderlinePosition -133
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/megies/git/obspy/obspy/core/utcdatetime.py", line 377, in __init__
    dt = datetime.datetime(*args, **kwargs)
TypeError: Required argument 'month' (pos 2) not found
```

I'm not sure it would help to see that message to be honest.

Reply:
In any case this PR is not making things worse. Before it there was a bare try/except, now at least I'm only catching an exception that is explicitly raised inside the subroutine that parses the timestamp.